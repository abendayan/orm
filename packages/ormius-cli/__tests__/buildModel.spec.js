const mockSelectOption = {
    init: jest.fn()
}

const mockUtils = {
    ensureDirSync: jest.fn(),
    capitalize: jest.fn(),
    FOLDERS: {
        MODEL: './models',
        MIGRATION: './migrations'
    }
}

const mockFs = {
    writeFileSync: jest.fn()
}

jest.mock('fs', () => (
    mockFs
))

const { generateModel, buildModel } = require('../cli/buildModel')

jest.mock('../cli/selectOption', () => (
    { selectOption: mockSelectOption }
))

jest.mock('../cli/utils', () => (
    mockUtils
))

describe('buildModel', () => {
    test('generateModel', () => {
        generateModel('test')
        expect(mockSelectOption.init.mock.calls).toMatchSnapshot()
    })

    test('buildModel', () => {
        Date.now = jest.fn(() => 1647098090769)
        buildModel([{ type: 'string', name: 'test' }], 'modelTest')
        expect(mockFs.writeFileSync.mock.calls).toMatchSnapshot()
        expect(mockUtils.capitalize.mock.calls).toMatchSnapshot()
        expect(mockUtils.ensureDirSync.mock.calls).toMatchSnapshot()
    })
})
