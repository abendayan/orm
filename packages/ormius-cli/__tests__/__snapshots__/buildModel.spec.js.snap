// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildModel buildModel 1`] = `
Array [
  Array [
    "./migrations/1647098090769/index.js",
    "const { Migration } = require('ormius')
class NewMigration extends Migration {

  constructor() {
    super()
    this.modelName='modelTest'
  }

  changes() {
    this.createTable([{\\"type\\":\\"string\\",\\"name\\":\\"test\\"}])
  }
}

module.exports = NewMigration",
    Object {
      "flag": "wx",
    },
  ],
  Array [
    "./models/modelTest/model.js",
    "const { TYPES } = require('ormius')
const model = {
  \\"test\\": {
    \\"type\\": TYPES.STRING
  },
}

const modelName = 'modelTest'

module.exports = {
    model, modelName
}",
    Object {
      "flag": "wx",
    },
  ],
  Array [
    "./models/modelTest/modelTest.js",
    "const { Model } = require('ormius')
const { TYPES } = require('ormius')
const { model, modelName } = require('./model')

class undefined extends Model {
  constructor(ormius) {
    const _model = {
      ...model,
    }
    super(modelName, _model, ormius.connection ? ormius.connection : ormius)
    this.model = model
  }
}

module.exports = {
  undefined
}",
    Object {
      "flag": "wx",
    },
  ],
]
`;

exports[`buildModel buildModel 2`] = `
Array [
  Array [
    "modelTest",
  ],
  Array [
    "modelTest",
  ],
]
`;

exports[`buildModel buildModel 3`] = `
Array [
  Array [
    "./models",
  ],
  Array [
    "./migrations",
  ],
  Array [
    "./migrations/1647098090769",
  ],
  Array [
    "./models/modelTest",
  ],
]
`;

exports[`buildModel generateModel 1`] = `
Array [
  Array [
    [Function],
    "test",
  ],
]
`;
